descartes_light/solvers/include/descartes_light/solvers/ompl/ompl_solver.h

```diff
@@ -0,0 +1,72 @@
+#ifndef DESCARTES_LIGHT_SOLVERS_OMPL_OMPL_SOLVER_H
+#define DESCARTES_LIGHT_SOLVERS_OMPL_OMPL_SOLVER_H
+
+#include <descartes_light/solvers/bgl/bgl_solver.h>
+#include <descartes_light/solvers/ompl/descartes_space.h>
+#include <ompl/base/Planner.h>
+#include <ompl/geometric/SimpleSetup.h>
+
+namespace descartes_light
+{
+/**
+ * @brief BGL solver implementation that constructs vertices and edges in the build function and uses Dijkstra's
+ * algorithm with a default visitor to search the graph
+ */
+template <typename FloatType>
+class BGLOMPLSolver : public BGLSolverBaseSVDE<FloatType>
+{
+public:
+  using BGLSolverBaseSVDE<FloatType>::BGLSolverBaseSVDE;
+  BGLOMPLSolver(double max_dist,
+                double planning_time,
+                unsigned num_threads = std::thread::hardware_concurrency(),
+                double rung_to_rung_dist = 100000)
+    : BGLSolverBaseSVDE<FloatType>(num_threads)
+    , max_dist_(max_dist)
+    , planning_time_(planning_time)
+    , rung_to_rung_dist_(rung_to_rung_dist)
+  {
+  }
+
+  SearchResult<FloatType> ompl_search(std::shared_ptr<ompl::base::Planner> ompl_planner);
+
+  void initOMPL();
+
+protected:
+  double max_dist_;
+  double planning_time_;
+  double rung_to_rung_dist_;
+  std::shared_ptr<descartes_light::DescartesStateSpace<FloatType>> dss_;
+  ompl::geometric::SimpleSetupPtr ss_ {nullptr};
+};
+/**
+ * @brief BGL solver implementation that constructs vertices and edges in the build function and uses Dijkstra's
+ * algorithm with a default visitor to search the graph
+ */
+template <typename FloatType>
+class BGLOMPLRRTSolver : public BGLOMPLSolver<FloatType>
+{
+public:
+  using BGLOMPLSolver<FloatType>::BGLOMPLSolver;
+
+  SearchResult<FloatType> search() override;
+};
+
+using BGLOMPLRRTSolverF = BGLOMPLRRTSolver<float>;
+using BGLOMPLRRTSolverD = BGLOMPLRRTSolver<double>;
+
+template <typename FloatType>
+class BGLOMPLRRTConnectSolver : public BGLOMPLSolver<FloatType>
+{
+public:
+  using BGLOMPLSolver<FloatType>::BGLOMPLSolver;
+
+  SearchResult<FloatType> search() override;
+};
+
+using BGLOMPLRRTConnectSolverF = BGLOMPLRRTConnectSolver<float>;
+using BGLOMPLRRTConnectSolverD = BGLOMPLRRTConnectSolver<double>;
```

Comment:
We should think about a way to make the specification of the search algorithm generic so we don't need to define new classes for every search algorithm we want to use. Depending on how exactly OMPL works, we could pass a search algorithm pointer into the constructor, or we could template this class on the type of search algorithm that should be used.

Reply:
One concern of mine is I'm not sure how many ompl algorithms will actually work with this setup, but I know RRT and RRTConnect do work. 

Your point still stands that it would be nice if one could just pass the ompl planner type. Unfortunately each planner has it's own unique set of parameters, so we would realistically need to account for every planner similar to how it's done in Tesseract. I think a better solution is to provide a method to `getSpaceInformation()` and `getMaxDistance()` from the base OMPLSolver class, which would allow a user to create the own instantiation of an ompl planner which then could be used to call `ompl_search`. It would look exactly like what is done in the `search()` method in `OMPLRRTSolver` and `OMPLRRTConnectSolver` but since it requires an input I think it should be kept separate from 'search()'